### ISSUE TITLE
Please do not coerce errors

### ISSUE THREAD

**Issue by snnsnn** [maverick-js/signals#28]
https://github.com/maverick-js/signals/blob/65f7fe9816595d3212130c3cc7d60986e6b8d346/src/core.ts#L268

Once we had a discussion on keeping the language semantics when handling errors in SolidJS. Solid did not allow throwing falsy values as errors even though the language permits it. I see you follow somewhat similar path, but coercing errors may not be good idea from the consumer's perspective.  Why not keep them as is:

```typescript
let handled = false;
for (let i = 0, len = handlers.length; i < len; i++) {
  try {
    handlers[i](error);
    handled = true;
    break; // Error is handled.
  } catch (rethrown: any) {
    error = rethrown;
  }
}
if (!handled) throw error;
```

Is there a valid reason for coercion?

**Comment by mihar-22** [comment]
Good point, let's remove it. Can't remember why but we shouldn't mess with it. Happy to pull in a PR. It's breaking change so we'll need to bump to v6.

**Comment by snnsnn** [comment]
> Good point, let's remove it. Can't remember why but we shouldn't mess with it. Happy to pull in a PR. It's breaking change so we'll need to bump to v6.

Ok. It is coming..
